docs/Modules/index.mdx

```diff
@@ -0,0 +1,106 @@
+
+Modules can be enabled using following methods:
+
+- During MSA deployment, a default validation module is enabled. Internally, it calls the [deployCounterFactualAccount](https://github.com/bcnmy/scw-contracts/blob/develop/contracts/smart-account/factory/SmartAccountFactory.sol#L57) method on smart account factory. The module selected to deploy the MSA determines its final address, due to its counterfactual nature.
+
+- A new module can be enabled via a userOp transaction on smart account by specifying the module address and moduleSetupData. Internally, it calls [setupAndEnableModule](https://github.com/bcnmy/scw-contracts/blob/develop/contracts/smart-account/base/ModuleManager.sol#L196) method on Module Manager. It can be set as an active validation module to be used for next transactions.
+
+```jsx
+const isEnabled = await smartAccount.isModuleEnabled(module_address)
+if ( !isEnabled ) {
+    const enableModuleTrx = await smartAccount.getEnableModuleData(module_address);
+    transactionArray.push( enableModuleTrx );
+}
+smartAccount = smartAccount.setActiveValidationModule( module )
+```
+
+As per the code, the initial step is to verify whether the module is already enabled. If the module is not enabled, a transaction can be performed to enable it.
+
+:::note
+The module passed during the deployment acts as default validaiton module. This module gets used as an active validaiton module, if no other module is enabled.
+:::
+
+## Validation Modules
+
+![validationModule.png](../images/modules/validationModule.png)
+
+The validation Module is a module that extends the abstract contract **BaseAuthorizationModule**** - which implements the **IBaseAuthorizationModule** interface. This interface requires the implementation of the following methods:
+
+- `validateUserOp`: this method validates userOperation. It expects userOp callData to execute method calls of the Smart account and userOp signature being the ABI-encoded signature and module address.
+- `isValidSignature`: this method validates an EIP-1271 signature.
+- `isValidSignatureUnsafe`: this method validates an EIP-1271 signature but expects the data Hash to already include smart account address information.
+
+Validation modules are invoked with a “call”. It has its own storage and doesn’t share storage with Biconomy Smart Account. Since validation module storage is accessed during the validation phase, ERC-4337 [storage rules](https://github.com/eth-infinitism/account-abstraction/blob/abff2aca61a8f0934e533d0d352978055fddbd96/eip/EIPS/eip-4337.md#storage-associated-with-an-address) apply to this.
+
+:::info
+Validation Modules in MSA play a crucial role in determining who can perform actions, ensuring security and proper authorization.
+:::
```

Comment:
```suggestion
The validation Module is a module that extends the abstract contract **BaseAuthorizationModule**** - which implements the **IBaseAuthorizationModule** interface. This interface requires the implementation of the following methods:

- `validateUserOp`: this method validates userOperation. It expects userOp callData to execute method calls of the Smart account and userOp signature being the ABI-encoded signature and module address.
- `isValidSignature`: this method validates an EIP-1271 signature.
- `isValidSignatureUnsafe`: this method validates an EIP-1271 signature but expects the data Hash to already include smart account address information.

Validation modules are invoked with a “call”. It has its own storage and doesn’t share storage with the Biconomy Smart Account. Since validation module storage is accessed during the validation phase, ERC-4337 [storage rules](https://github.com/eth-infinitism/account-abstraction/blob/abff2aca61a8f0934e533d0d352978055fddbd96/eip/EIPS/eip-4337.md#storage-associated-with-an-address) apply to this.

:::info
Validation Modules in MSA play a crucial role in determining who can perform actions, ensuring security and proper authorization.
:::
```